lib/gooddata/bricks/middleware/aws_middleware.rb

```diff
@@ -12,21 +12,48 @@ module Bricks
     class AWSMiddleware < Bricks::Middleware
       def call(params)
         params = params.to_hash
-        if params.key?('aws_client')
+        s3_config = get_s3_config(params)
+
+        unless s3_config.empty?
           GoodData.logger.info('Setting up AWS-S3 connection')
-          raise 'Unable to connect to AWS. Parameter "aws_client" seems to be empty' unless params['aws_client']
-          raise 'Unable to connect to AWS. Parameter "access_key_id" is missing' if params['aws_client']['access_key_id'].blank?
-          raise 'Unable to connect to AWS. Parameter "secret_access_key" is missing' if params['aws_client']['secret_access_key'].blank?
-          params['aws_client'] = rewrite_for_aws_sdk_v2(params['aws_client'])
-          symbolized_config = GoodData::Helpers.symbolize_keys(params['aws_client'])
-          s3 = Aws::S3::Resource.new(symbolized_config)
-          params['aws_client']['s3_client'] = s3
+          if params.key?('aws_client')
+            params['s3_client'] = {}
+          elsif params['input_source'] && params['input_source']['bucket'].blank?
+            if params['s3_client']['bucket'].present?
+              params['input_source']['bucket'] = params['s3_client']['bucket']
+            else
+              GoodData.logger.warn('Parameter "bucket" is missing, you may need to specify it in "s3_client" or "input_source"')
+            end
+          end
+          s3_config = rewrite_for_aws_sdk_v2(s3_config)
+          symbolized_config = GoodData::Helpers.symbolize_keys(s3_config)
+          params['s3_client']['client'] = Aws::S3::Resource.new(symbolized_config)
         end
         @app.call(params)
       end
 
       private
 
+      def get_s3_config(params)
+        s3_config = {}
+        if params.key?('aws_client')
```

Comment:
We should check if both aws_client and s3_client existing and log warning message: "Found two configuration aws_client and s3_client for S3 input source, use aws_client configuration".

Reply:
Added this log. Thanks